neurom/core/_neuron.py

```diff
@@ -217,6 +227,12 @@ def __deepcopy__(self, memo):
     def __nonzero__(self):
         return bool(self.root_node)
 
+    def __eq__(self, other):
```

Comment:
do you need `__eq__` and `__hash__` in this review?

Reply:
I needed __eq__ for the assert_equal of the unit tests.
And apparently you need to implement __hash__ whenever you implement __eq__: 
https://stackoverflow.com/questions/1608842/types-that-define-eq-are-unhashable